inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// This pass changes geometry of convolution stages in order
+// to get more efficient HW tiling (pass "hwConvTiling") using reshape stages.
+
+#include <vpu/middleend/pass_manager.hpp>
+
+namespace vpu {
+
+namespace {
+
+class PassImpl final : public Pass {
+public:
+    explicit PassImpl(const StageBuilder::Ptr& stageBuilder) : _stageBuilder(stageBuilder) {}
+
+    void run(const Model& model) override;
+
+private:
+    StageBuilder::Ptr _stageBuilder;
+};
+
+void PassImpl::run(const Model& model) {
+    VPU_PROFILE(hwConvTiling);
+    for (const auto& stage : model->getStages()) {
+        if (stage->type() != StageType::StubConv) {
+            continue;
+        }
+
+        const auto tryHW = stage->attrs().getOrDefault<bool>("tryHW", false);
+        if (!tryHW) {
+            continue;
+        }
+
+        if (stage->attrs().get<int>("kernelSizeX") != 1 ||
+            stage->attrs().get<int>("kernelSizeY") != 1) {
+            continue;
+        }
+
+        const auto input = stage->input(0);
+        const auto output = stage->output(0);
+
+        const auto& inputDesc = input->desc();
+        const auto& outputDesc = output->desc();
+
+        if ((inputDesc.dimsOrder() != DimsOrder::NCHW) ||
+            (outputDesc.dimsOrder() != DimsOrder::NCHW)) {
+            continue;
+        }
+
+        const auto actualInputShape = inputDesc.dims();
+        const auto expectedInputShape = DimValues{ {Dim::N, 1}, {Dim::C, 608}, {Dim::H, 34}, {Dim::W, 60} };
+
+        if (actualInputShape != expectedInputShape) {
+            continue;
+        }
+
+        if (outputDesc.dim(Dim::W) != inputDesc.dim(Dim::W) ||
+            outputDesc.dim(Dim::H) != inputDesc.dim(Dim::H)) {
```

Comment:
I think we should check strides == 1 for this pass. And throw if dims are not equal

Reply:
I added check strides == 1. I removed check if dims not equal, because I don't think this pass should provide information about incorrectness of structure of the network